packages/@ember/object/type-tests/set.test.ts

```diff
@@ -0,0 +1,14 @@
+import { set } from '@ember/object';
+
+import { expectTypeOf } from 'expect-type';
+
+let foo = { baz: 1 };
+
+expectTypeOf(set(foo, 'baz', 2)).toEqualTypeOf<number>();
+
+// FIXME: This is definitely a potential footgun. You're allowed to
+// change the type without any warning.
+expectTypeOf(set(foo, 'baz', 'aaaahhh')).toEqualTypeOf<string>();
```

Comment:
Yeah, this is why we originally required the `T[K]` type here. My current thinking is that maybe we *do* support that for standalone `get` and `set`, *not* out of the box but via the extension types? But with an eye to deprecating both post-Polaris. Hmm.

Reply:
Sounds like we should discuss this some more.